aws/resource_aws_api_gateway_domain_name_test.go

```diff
@@ -348,17 +341,64 @@ func testAccCheckAWSAPIGatewayDomainNameDestroy(s *terraform.State) error {
 	return nil
 }
 
-func testAccAWSAPIGatewayDomainNameConfig_CertificateArn(domainName, certificateArn string) string {
-	return fmt.Sprintf(`
+func testAccAWSAPIGatewayDomainNameConfig_CertificateArn(rootDomain string, domain string) string {
+	return composeConfig(
+		testAccApigatewayEdgeDomainNameRegionProviderConfig(),
+		fmt.Sprintf(`
+data "aws_route53_zone" "test" {
+  name         = %[1]q
+  private_zone = false
+}
+
+resource "aws_acm_certificate" "test" {
+  domain_name       = %[2]q
+  validation_method = "DNS"
+}
+
+#
+# for_each acceptance testing requires:
+# https://github.com/hashicorp/terraform-plugin-sdk/issues/536
+#
+# resource "aws_route53_record" "test" {
+#   for_each = {
+#     for dvo in aws_acm_certificate.test.domain_validation_options: dvo.domain_name => {
+#       name   = dvo.resource_record_name
+#       record = dvo.resource_record_value
+#       type   = dvo.resource_record_type
+#     }
+#   }
+
+#   allow_overwrite = true
+#   name            = each.value.name
+#   records         = [each.value.record]
+#   ttl             = 60
+#   type            = each.value.type
+#   zone_id         = data.aws_route53_zone.test.zone_id
+# }
+
+resource "aws_route53_record" "test" {
+  allow_overwrite = true
+  name            = tolist(aws_acm_certificate.test.domain_validation_options)[0].resource_record_name
+  records         = [tolist(aws_acm_certificate.test.domain_validation_options)[0].resource_record_value]
+  ttl             = 60
+  type            = tolist(aws_acm_certificate.test.domain_validation_options)[0].resource_record_type
+  zone_id         = data.aws_route53_zone.test.zone_id
+}
+
+resource "aws_acm_certificate_validation" "test" {
+  certificate_arn         = aws_acm_certificate.test.arn
+  validation_record_fqdns = [aws_route53_record.test.fqdn]
+}
```

Comment:
FYI, in https://github.com/hashicorp/terraform-provider-aws/pull/15258 I have pulled this into a separate function as I will use it in a couple of places: https://github.com/hashicorp/terraform-provider-aws/pull/15258/commits/4a7197b8d5be6b44fc3d11dbf85f1baf6b69ce83#diff-de07b61dfd6679fbda71d2c219efc2356d9a0757a6366304a5361ee09f86803bR395-R441.
If this PR is merged first I'll refactor `testAccAWSAPIGatewayDomainNameConfig_CertificateArn` to use this new function.

Reply:
Okay cool -- feel free to add it to the contributing guide and create a followup technical debt issue. 👍

Reply:
Done: https://github.com/hashicorp/terraform-provider-aws/issues/16171.